core-v1/src/main/scala/data/ExtractValue.scala

```diff
@@ -0,0 +1,26 @@
+package com.rallyhealth.vapors.v1
+
+package data
+
+/**
+  * Extracts a value from some starting value.
+  */
+trait ExtractValue[-T, +V] {
```

Comment:
I should probably avoid variance here... most libraries that define typeclasses like this are invariant.